arch/risc-v/src/common/riscv_macros.S

```diff
@@ -106,11 +106,6 @@
   and        t2, t0, t1
   li         t1, MSTATUS_FS_DIRTY
   bne        t2, t1, 1f
-  li         t1, ~MSTATUS_FS
-  and        t0, t0, t1
-  li         t1, MSTATUS_FS_CLEAN
-  or         t0, t0, t1
-  REGSTORE   t0, REG_INT_CTX(\in)
```

Comment:
You should still propagate the state to CLEAN, as per the riscv privileged spec.

Reply:
https://github.com/apache/nuttx/commit/45bc40d52c8ff9a892af8c2f069dacf12baa4dd2